src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -634,24 +635,42 @@ ModifyPartialQuerySupported(Query *queryTree, bool multiShardQuery,
 		foreach(targetEntryCell, queryTree->targetList)
 		{
 			TargetEntry *targetEntry = (TargetEntry *) lfirst(targetEntryCell);
+
+			/* skip resjunk entries: UPDATE adds some for ctid, etc. */
+			if (targetEntry->resjunk)
+			{
+				continue;
+			}
+
 			bool targetEntryPartitionColumn = false;
+			AttrNumber attNumber = 0;
```

Comment:
minor: use `InvalidAttrNumber`

Also, `attNunber` is way too generic here making it hard to follow. Can you rename throughout the PR as `targetColumnAttrNumber` or such?